plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcMetadataFactory.java

```diff
@@ -14,28 +14,31 @@
 package io.trino.plugin.jdbc;
 
 import io.airlift.units.Duration;
+import io.trino.plugin.base.session.SessionPropertiesProvider;
 
 import javax.inject.Inject;
 
-import java.util.concurrent.TimeUnit;
+import java.util.Set;
 
 import static java.util.Objects.requireNonNull;
 
 public class JdbcMetadataFactory
 {
     private final JdbcClient jdbcClient;
+    private final Set<SessionPropertiesProvider> sessionPropertiesProviders;
     private final boolean allowDropTable;
 
     @Inject
-    public JdbcMetadataFactory(JdbcClient jdbcClient, JdbcMetadataConfig config)
+    public JdbcMetadataFactory(JdbcClient jdbcClient, JdbcMetadataConfig config, Set<SessionPropertiesProvider> sessionPropertiesProviders)
     {
         this.jdbcClient = requireNonNull(jdbcClient, "jdbcClient is null");
+        this.sessionPropertiesProviders = requireNonNull(sessionPropertiesProviders, "sessionPropertiesProviders is null");
         requireNonNull(config, "config is null");
         this.allowDropTable = config.isAllowDropTable();
     }
 
     public JdbcMetadata create()
     {
-        return new JdbcMetadata(new TransactionCachingJdbcClient(jdbcClient, new Duration(1, TimeUnit.DAYS)), allowDropTable);
+        return new JdbcMetadata(new CachingJdbcClient(jdbcClient, sessionPropertiesProviders, Duration.valueOf("1d"), true), allowDropTable);
```

Comment:
In the removed class we had

```
 // session stays the same per transaction, therefore session properties don't need to
// be a part of cache keys in CachingJdbcClient
```

i think we can keep that logic 

- use empty Set instead of sessionPropertiesProviders
- keep the comment